apps/prairielearn/src/middlewares/studentAssessmentAccess.html.ts

```diff
@@ -0,0 +1,90 @@
+  resLocals,
+  showClosedScore = true,
+}: {
+  resLocals: Record<string, any>;
+  showClosedScore?: boolean;
+}) {
+  const { assessment, assessment_set, assessment_instance, authz_result } = resLocals as {
+    assessment: Assessment;
+    assessment_set: AssessmentSet;
+    assessment_instance?: AssessmentInstance;
+    authz_result: any;
+  };
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${renderEjs(import.meta.url, "<%- include('../pages/partials/head') %>", resLocals)}
+      </head>
+      <body>
+        ${renderEjs(import.meta.url, "<%- include('../pages/partials/navbar'); %>", {
+          ...resLocals,
+          navPage: '',
+        })}
+        <main id="content" class="container">
+          <div class="card mb-4">
+            <div class="card-header bg-primary text-white">
+              ${assessment_set.abbreviation}${assessment.number}: ${assessment.title}
+            </div>
+
+            <div class="card-body">
+              ${assessment_instance != null &&
+              (assessment_instance.open === false || authz_result.active === false) &&
+              showClosedScore
+                ? html`
+                    <div class="row align-items-center">
+                      <div class="col-md-3 col-sm-6">
+                        Total points:
+                        ${formatPoints(assessment_instance.points)}/${formatPoints(
+                          assessment_instance.max_points,
+                        )}
+                      </div>
+                      <div class="col-md-3 col-sm-6">
+                        ${renderEjs(
+                          import.meta.url,
+                          "<%- include('../pages/partials/scorebar') %>",
+                          { score: assessment_instance.score_perc },
+                        )}
+                      </div>
+
+                      ${AssessmentStatusDescription({
+                        assessment_instance,
+                        authz_result,
+                        extraClasses: 'col-md-6 col-sm-12 text-right',
+                      })}
+                    </div>
+                  `
+                : AssessmentStatusDescription({ assessment_instance, authz_result })}
+            </div>
+            <!-- panel-body -->
+          </div>
+        </main>
+      </body>
+    </html>
+  `.toString();
+}
+
+function AssessmentStatusDescription({
+  assessment_instance,
+  authz_result,
+  extraClasses = '',
+}: {
+  assessment_instance?: AssessmentInstance;
+  authz_result: any;
+  extraClasses?: string;
+}) {
+  return html`<div class="test-suite-assessment-closed-message ${extraClasses}">
+    ${assessment_instance?.open === false
+      ? html`Assessment is <strong>closed</strong> and is no longer available.`
+      : authz_result.next_active_time == null
+        ? html`Assessment is no longer available.`
+        : html`Assessment will become available on ${authz_result.next_active_time}.`}
+  </div>`;
```

Comment:
This will probably need prettier:

```suggestion
  return html`
  <div class="test-suite-assessment-closed-message ${extraClasses}">
    ${assessment_instance?.open === false
      ? html`Assessment is <strong>closed</strong> and is no longer available.`
      : authz_result.next_active_time == null
        ? html`Assessment is no longer available.`
        : html`Assessment will become available on ${authz_result.next_active_time}.`}
  </div>
  `;
```

Reply:
5df752d9b